src/cpp/session/modules/SessionDataViewer.R

```diff
@@ -414,22 +414,28 @@
   }
 
   # apply sort
-  if (col > 0 && length(x[[col]]) > 0)
+  if (length(cols) > 0)
   {
-    if (is.list(x[[col]][[1]]) || length(x[[col]][[1]]) > 1)
+    vals <- list()
+    for (i in length(cols))
     {
-      # extract the first value from each cell for ordering (handle
-      # vector-valued columns gracefully)
-      x <- as.data.frame(x[order(vapply(x[[col]], `[`, 0, 1), 
-                                 decreasing = identical(dir, "desc")), ,
-                           drop = FALSE])
+      idx <- cols[[i]]
+      if (length(x[[idx]]) > 0)
+      {
+        if (dirs[[i]] == "asc")
```

Comment:
Because of R's wonky comparison semantics, use `identical(x, y)` to compare strings instead of `==` (for pretty much the same reasons `===` is generally preferred over `==` in JS)

Reply:
The most famous example:

```
> x <- NULL
> y <- "foo"
> if (x == y) message("Equality achieved")
Error in if (x == y) message("Equality achieved") : 
  argument is of length zero
```